src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs

```diff
@@ -15,7 +16,16 @@ public static class SentryOptionsDiagnosticExtensions
         /// </summary>
         /// <param name="options">The Sentry options.</param>
         public static void AddDiagnosticSourceIntegration(this SentryOptions options)
-            => options.AddIntegration(new SentryDiagnosticListenerIntegration());
+        {
+            if (options.Integrations != null &&
+                options.Integrations.OfType<SentryDiagnosticListenerIntegration>().Any())
```

Comment:
@bruno-garcia @mattjohnsonpint  should i refactor this into a helper method in options?

Reply:
I don't think that's necessary.